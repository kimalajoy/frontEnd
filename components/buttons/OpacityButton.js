import React from 'react';
import { TouchableOpacity, Text } from 'react-native'

import styles from '../../styles/styles'

export const OpacityButton = ({buttonStyle, pressFunction, text, textStyle}) => {
  return(
    <TouchableOpacity
      style={ buttonStyle }
      onPress={ pressFunction }
      >
      <Text style={ textStyle }>
        {text}
      </Text>
    </TouchableOpacity>
  )
}
