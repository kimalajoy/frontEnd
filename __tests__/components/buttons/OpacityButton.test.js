import React from 'react';
import { render, cleanup, fireEvent } from 'react-native-testing-library';

import { OpacityButton } from '../../../components/buttons/OpacityButton';

describe("TouchableOpacity Button", () => {
  let button;

  beforeEach(() => {
    onPressMock = jest.fn();

    button = render(<OpacityButton
                      text={"Test Text"}
                      pressFunction={onPressMock}/>);
  });

  afterEach(cleanup)

  test("renders the passed text", () => {
    const thisButton = button.queryByText('Test Text');

    expect(thisButton).not.toBeNull();
  })

  test('will fire the onPress event', () => {
    const thisButton = button.getByText('Test Text');

    fireEvent.press(thisButton);
    expect(onPressMock).toHaveBeenCalled();
  })

  test('will render as expected', () => {
    expect(button.toJSON()).toMatchSnapshot()
  })

})
