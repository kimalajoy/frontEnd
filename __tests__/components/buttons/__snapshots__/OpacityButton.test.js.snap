// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TouchableOpacity Button will render as expected 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text>
    Test Text
  </Text>
</View>
`;
